docassemble/AssemblyLine/test_al_document.py

```diff
@@ -0,0 +1,37 @@
+import unittest
+from docassemble.base.util import DAFile
+from .al_document import ALDocument, ALDocumentBundle
+
+
+class test_dont_assume_pdf(unittest.TestCase):
+    def test_upload_pdf(self):
```

Comment:
I'm not sure I understand what this is testing.

The original error that prompted this PR was that I assumed:

1. The contents of the bundle were generated by an attachment block that produced a [DAFileCollection](https://docassemble.org/docs/objects.html#DAFileCollection) object
2. If the .docx attribute didn't exist for the DAFileCollection object, the .pdf attribute would exist

But this assumption failed if it wasn't a DAFileCollection, or if the filetype wasn't convertable to PDF and wasn't originally a DOCX.

But this could still be fine as-is and we can open a new issue to review and add unit tests

Reply:
Gotcha, I never realized that attachment blocks were DAFileCollections. I do think this PR handles "if it wasn't a DAFileCollection", but I see what you mean, probably should have tested with a jpg or some other file type.